tests/analyze-fixtures/VariableVariableUsage.php

```diff
@@ -7,14 +7,63 @@ class VariableVariableUsage
     /**
      * @return string
      */
-    public function method()
+    public function simpleVariableVariable()
     {
         $varName = 'name';
 
         ${$varName} = 'foo';
 
         return $name;
     }
+
+    /**
+     * @return string
+     */
+    public function variableVariableInArrayAssignment()
+    {
+        $array = [];
+        $varName = 'array';
+
+        ${$varName}[] = 'foo';
+        $array[] = 'bar';
+
+        return $array;
+    }
+
+    /**
+     * @return string
+     */
+    public function variableVariableInListAssignment()
+    {
+        $a = 'foo';
+        list(${$a}, $b) = [1, 2];
+
+        return $foo;
+    }
+
+    /**
+     * @return string
+     */
+    public function propertyAccessVariable()
+    {
+        $varName = 'name';
+
+        $this->{$varName} = 'foo';
+
+        return $name;
+    }
+
+    /**
+     * @return string
+     */
+    public function arrayPropertyAccessVariable()
```

Comment:
It would be best if we had one test for each case where it doesn't create a notice.